deploy/testenv/docker-compose.yaml

```diff
@@ -131,6 +138,72 @@ services:
     deploy:
       replicas: 1
 
+
+  #  Real-time Dashboards
+  #  ------------------------------
+  grafana:
+    depends_on:
+      - loki
+    image: grafana/grafana:10.2.2
+    ports:
+      - "3000:3000"
+    environment:
+      GF_AUTH_ANONYMOUS_ENABLED: true
+      GF_AUTH_ANONYMOUS_ORG_ROLE: Admin
+      GF_PATHS_PROVISIONING: '/app.cfg/provisioning'
+    volumes:
+        - ./grafana:/app.cfg
+
+
+  #  Logging with Loki
+  #  ------------------------------
+  loki:
+      image: grafana/loki:2.9.0
+      ports:
+        - "3100:3100"
+      volumes:
+        - ./loki:/etc/loki
+      command: -config.file=/etc/loki/loki-local-config.yaml
+
+  promtail:
+      depends_on:
+          - loki
+      image: grafana/promtail:2.9.0
+      volumes:
+          # Configuration
+          - ./promtail:/etc/promtail
+          # Read container labels and logs
+          - /var/lib/docker/containers:/var/lib/docker/containers:ro
+          - /var/run/docker.sock:/var/run/docker.sock
+      command: -config.file=/etc/promtail/docker-config.yaml
+
+
+  #  Monitoring with Prometheus
+  #  ------------------------------
+  prometheus:
+    image: prom/prometheus:v2.7.1
+    ports:
+      - "9090:9090"
+    volumes:
+      - ./prometheus:/app.cfg
+    command: >-
+      --config.file=/app.cfg/prometheus.yml
+      --storage.tsdb.path=/prometheus
+      --web.console.libraries=/usr/share/prometheus/console_libraries
+      --web.console.templates=/usr/share/prometheus/consoles
+
+  cadvisor:
```

Comment:
What does this do?

Reply:
some optional stuff for the web interface of Prometheus. Can be removed.